lib/DxcSupport/dxcapi.extval.cpp

```diff
@@ -0,0 +1,58 @@
+#include "dxc/Support/WinIncludes.h"
+#include <dxc/Support/Global.h> // for hresult handling with DXC_FAILED
+#include <filesystem>           // C++17 and later
+// WinIncludes must come before dxcapi.extval.h
+#include "dxc/Support/dxcapi.extval.h"
+
+namespace dxc {
+
+HRESULT DxcDllExtValidationSupport::CreateInstance(REFCLSID clsid, REFIID riid,
+                                                   IUnknown **pResult) {
+  if (DxilExtValSupport.IsEnabled() && clsid == CLSID_DxcValidator)
+    return DxilExtValSupport.CreateInstance(clsid, riid, pResult);
+
+  return DxcompilerSupport.CreateInstance(clsid, riid, pResult);
+}
+
+HRESULT DxcDllExtValidationSupport::CreateInstance2(IMalloc *pMalloc,
+                                                    REFCLSID clsid, REFIID riid,
+                                                    IUnknown **pResult) {
+  if (DxilExtValSupport.IsEnabled() && clsid == CLSID_DxcValidator)
+    return DxilExtValSupport.CreateInstance2(pMalloc, clsid, riid, pResult);
+
+  return DxcompilerSupport.CreateInstance2(pMalloc, clsid, riid, pResult);
+}
+
+HRESULT DxcDllExtValidationSupport::InitializeInternal(LPCSTR dllName,
+                                                       LPCSTR fnName) {
+  // Load dxcompiler.dll
```

Comment:
If we're certain that this will _only_ ever be used to load dxcompiler.dll then maybe dllName shouldn't be an argument?